taichi/program/sparse_matrix.cpp

```diff
@@ -57,6 +57,20 @@ void print_triplet_from_csr(int64_t n_rows,
   ostr << Eigen::MatrixXf(m.cast<float>()).format(clean_fmt);
 }
 
+template <typename T, typename T1, typename T2>
+T2 get_element_from_csr(int row,
+                        int col,
+                        T *row_data,
+                        T1 *col_data,
+                        T2 *value) {
+  for (T i = row_data[row]; i < row_data[row + 1]; ++i) {
```

Comment:
Is `T` and `T1` always `int`?

Reply:
Good point, let me simply set them to `int`. Thanks.

Reply:
Good point, let me simply set them to `int`. Thanks!

Reply:
Thanks and Good Point! 
My thought is: Although we only allow `int` to be the type of `row` & `col`, the `cuSparse` library uses both `int32` & `int64` as the types. For the convenience of future development where we might involve more of `cuSparse` in Taichi, I leave different type names here. : )